src/Featurizers/UnitTests/LagLeadOperatorFeaturizer_UnitTests.cpp

```diff
@@ -0,0 +1,270 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include "../LagLeadOperatorFeaturizer.h"
+#include "../TestHelpers.h"
+
+namespace NS = Microsoft::Featurizer;
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+TEST_CASE("Invalid constructor parameter") {
```

Comment:
>TEST_CASE [](start = 0, length = 9)

I would expect to see many more tests given the complexity of the code. There should be tests that exercise each branch in the logic.